keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -73,6 +75,21 @@ class ApplicationService(
         judgedBy = user))
   }
 
+  fun pin(application: String, pin: EnvironmentArtifactPin, user: String) {
+    val config = repository.getDeliveryConfigForApplication(application)
```

Comment:
can the `config` here and below be null/empty? are we throwing an exception in those cases?

Reply:
Good question. Feels like a 404 would make sense in that case.

Reply:
I can, and `repository.getDeliveryConfigForApplication` will throw an exception (should be 404)